Mathlib/Tactic/MoveAdd.lean

```diff
@@ -439,7 +443,7 @@ elab (name := moveOperTac) "move_oper" id:ident rws:rwRuleSeq : tactic => do
   -- parse the operation
   let op := id.getId
   -- parse the list of terms
-  let (instr, (unmatched, stxs), dbgMsg) ← unifyMovements op (← parseArrows rws)
+  let (instr, (unmatched, stxs), dbgMsg) ← unifyMovements op (← getMainGoal) (← parseArrows rws)
```

Comment:
While setting the context is surely the bug, I think the simpler solution is to do `... : tactic => withMainContext do` on the `elab` line.

That fixes another bug: `parseArrows` uses `elabTerm`, and that assumes the context is set correctly.

With that modification, you don't need to thread the main goal into either `unifyMovements` or `pairUp`.